hollow/src/main/java/com/netflix/hollow/tools/stringifier/HollowRecordStringifier.java

```diff
@@ -67,60 +69,70 @@ public HollowRecordStringifier addExcludeObjectTypes(String... types) {
     }
 
 
-    /**
-     * create a String representation of the specified {@link HollowRecord}.
-     */
     @Override
     public String stringify(HollowRecord record) {
-        return stringify(record.getTypeDataAccess().getDataAccess(), record.getSchema().getName(), record.getOrdinal());
+        return stringify(record.getTypeDataAccess().getDataAccess(),
+                record.getSchema().getName(), record.getOrdinal());
     }
 
-    /**
-     * create a String representation of the record in the provided dataset, of the given type, with the specified ordinal.
-     */
     @Override
-    public String stringify(HollowDataAccess dataAccess, String type, int ordinal) {
-        StringBuilder builder = new StringBuilder();
+    public void stringify(Writer writer, HollowRecord record) throws IOException {
+        stringify(writer, record.getTypeDataAccess().getDataAccess(), record.getSchema().getName(),
+                record.getOrdinal());
+    }
 
-        appendStringify(builder, dataAccess, type, ordinal, 0);
+    @Override
+    public String stringify(HollowDataAccess dataAccess, String type, int ordinal) {
+        try {
+            StringWriter writer = new StringWriter();
+            stringify(writer, dataAccess, type, ordinal);
+            return writer.toString();
+        } catch (IOException e) {
+            throw new IllegalStateException("Unexpected exception using StringWriter", e);
```

Comment:
I tend to reserve `IllegalStateException` for when a class has gotten into a bad internal state, e.g. where assumptions or invariants are violated. For demoting a checked exception I often use `RuntimeException` unless a better subclass presents itself.

Reply:
Fair enough, updated